src/BSON/Binary.c

```diff
@@ -142,6 +152,32 @@ static void php_phongo_binary_free_object(void *object TSRMLS_DC) /* {{{ */
 	efree(intern);
 } /* }}} */
 
+zend_object* php_phongo_binary_create_object(zend_class_entry *class_type TSRMLS_DC) /* {{{ */
+{
+        php_phongo_binary_t *intern;
+
+        intern = (php_phongo_binary_t *)ecalloc(1, sizeof(php_phongo_binary_t)+zend_object_properties_size(class_type));
+
+        zend_object_std_init(&intern->std, class_type TSRMLS_CC);
+        object_properties_init(&intern->std, class_type);
+
+        intern->std.handlers = &php_phongo_handler_binary;
+
+        return &intern->std;
+} /* }}} */
+#else
+static void php_phongo_binary_free_object(void *object TSRMLS_DC) /* {{{ */
+{
```

Comment:
oboy. If the only thing in these two free handlers is void* vs zend_object* which we then pass to Z_BINARY_OBJ_P() to get the php_phongo_binary_t*....

How about adding in phongo_compat.h a `phongo_free_object_arg` which is defined as void or zend_object?
Then we don't have to duplicate any of the free_object implementations at all.
Its a bit fugly, but I think its better then duplicating loads of code that risks diverging in the future